### PR TITLE
Changes behavior of blocks.

### LINKED ISSUE #967: Blocks: allow to choose any maximum number of moves
From @maxamine

Currently, blocks properties allow to set (no movement / 1 movement only / unlimited movements).
it would be nice if we could set a maximum number of movements we want, we could use it to create some nice puzzles for example.

### PR DESCRIPTION
**Pull request by Nate-Devv**
Allows them to be pushed a multiple, finite number of times.
Infinite pushes has become -1 instead of 2.

Closes #967.

---
This change is [Reviewable](https://reviewable.io/reviews/christopho/solarus/976)

### CODE REVIEW

**Comment by christopho** [comment]
Thanks! I put a comment in the issue to keep track of what remains to be done.

**Comment by Nate-Devv** [comment]
Ok. I'll look at getting the rest of the changes in.

**Comment by Diarandor** [comment]
@Nate-Devv: You say infinite pushes has become -1 instead of 2. However, for infinite pushes it was not 2 but nil (that is what the API of Solarus v1.4 and v1.5 say)
If we set the max number of pushes to nil, will it still mean infinite?

**Comment by Diarandor** [comment]
Ok, I guess that you refer to the variables in the C++ side, which may have different values, so in that case there is no problem. Otherwise the API would be wrong.

**Comment by Nate-Devv** [comment]
I was talking about the C++ side, currently working on the lua and doc side.

**Comment by Diarandor** [comment]
Ok, sorry!

**Comment by Nate-Devv** [comment]
No problem!

File: src/lua/MapApi.cpp [hunk 1]
```diff
@@ -778,7 +778,7 @@ int LuaContext::l_create_block(lua_State* l) {
     EntityData& data = *(static_cast<EntityData*>(lua_touserdata(l, 2)));
```
Comment (Nate-Devv) [inline src/lua/MapApi.cpp]: Is this sending a lua nil to a value of two? If so, it should be changed to `EntityData& data = *(static_cast<EntityData*>(lua_touserdata(l, 2)));`

**Comment by christopho** [comment]
Merged into a feature branch blocks-any-maximum. Thanks!

**Pull request closed without merging**